src/internal/analytics/__tests__/components/analytics-funnel.test.tsx

```diff
@@ -641,6 +641,48 @@ describe('AnalyticsFunnelSubStep', () => {
   });
 });
 
+describe('emit cloudscape-creation-flow-done event', () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+    jest.useFakeTimers();
+    mockFunnelMetrics();
+  });
+
+  test('emit cloudscape-creation-flow-done event when funnelComplete', async () => {
+    let eventIsCalled = false;
+    document.addEventListener('cloudscape-creation-flow-done', () => {
+      eventIsCalled = true;
+    });
+    const ChildComponent = () => {
+      const { funnelSubmit } = useFunnel();
+      return <button onClick={funnelSubmit}>Submit</button>;
+    };
+    const { getByText } = render(
+      <AnalyticsFunnel funnelType="single-page" optionalStepNumbers={[]} totalFunnelSteps={3}>
+        <ChildComponent />
+      </AnalyticsFunnel>
+    );
+    act(() => void jest.runAllTimers());
+    fireEvent.click(getByText('Submit'));
+    await waitFor(() => {
+      expect(eventIsCalled).toBe(true);
```

Comment:
Adding to my previous comment, you can probably turn the event listener into a jest function and then do `expect(...).toHaveBeenCalled` here. Seems a bit more like standard jest testing